examples/build_all.py

```diff
@@ -1,29 +1,46 @@
 #!/usr/bin/python3
 
+import os
+
 from pathlib import Path
 from subprocess import PIPE, run
 
+# NOTE: boost V1.67 is to old! CK
 examples = [
-  x for x in Path(__file__).parent.iterdir() if x.is_dir() and (x / 'CMakeLists.txt').exists()
+    x for x in Path(__file__).parent.iterdir() if x.is_dir() and (x / 'CMakeLists.txt').exists() and (not x.name in ['boost', 'old-gtest'])
```

Comment:
On my system (Mac OS 10.15.7, Apple clang 12.0.0) both the boost and gtest examples compile without issues. Which versions are you using and what problems are they creating? Also not sure the `old-gtest` directory exists in the repo.

Tbh I'd rather try to fix the example or remove them completely, if unfixable on modern systems instead of excluding them from the script / CI system.

Reply:
it seems, that the build is not independent of the libraries installed on my build host: 

bash-3.2$ brew list
abseil			diffutils		git-flow		jpeg			libtool			openjdk			ruby
adns			docbook			git-gui			jsoncpp			libunistring		openldap		smartmontools
asciidoc		docker			glib			lcov			libusb			openssl@1.1		snappy
astyle			doctest			glog			libassuan		libyaml			p11-kit			source-highlight
autoconf		dos2unix		gmp			libcbor			llvm			pango			spdlog
automake		double-conversion	gnupg			libev			lua			pcre			sqlite
bash-completion		doxygen			gnutls			libevent		lz4			pcre2			swig
bdw-gc			duplicity		go			libffi			lzo			pinentry		tcl-tk
boost			findutils		gobject-introspection	libfido2		make			pixman			tidy-html5
brotli			fmt			googletest		libgcrypt		meson			pkg-config		tree
c-ares			folly			graphite2		libgpg-error		mpfr			poco			unbound
cairo			fontconfig		graphviz		libidn2			multimarkdown		popt			webp
ccache			freetype		grpc			libksba			multitail		protobuf		wget
cereal			fribidi			gts			libmetalink		ncurses			pylint			xmake
cmake			gcc			guile			libmpc			net-snmp		python@3.9		xz
conan			gd			harfbuzz		libpng			netpbm			range-v3		yamllint
coreutils		gdbm			icu4c			librsvg			nettle			rapidjson		zstd
cpp-gsl			gdk-pixbuf		isl			librsync		nghttp2			re2
cppcheck		gettext			jasper			libssh2			ninja			readline
ctags			gflags			jemalloc		libtasn1		npth			renameutils
curl			git			jinja2-cli		libtiff			omniorb			rtmpdump
bash-3.2$

Reply:
do you have a solution to get boost v1.75?

Reply:
Unresolving, as the point above has not been addressed.

> do you have a solution to get boost v1.75?

So far the only CMake compatible version I've seen is [Orphis/boost-cmake](https://github.com/Orphis/boost-cmake). For me and on the CI it hasn't been creating any issues, so I see no problem with keeping it until we can replace it with a newer version.

Also note that `old-gtest` does not seem to exist in this directory, so I don't see the purpose of the exclusion.

Reply:
done